src/compiler/utilities.ts

```diff
@@ -1655,8 +1655,9 @@ namespace ts {
      * @param location An optional TextRange to use to supply the new position.
      * @param flags The NodeFlags to use for the cloned node.
      * @param parent The parent for the new node.
+     * @param original An optional pointer to the original node.
      */
-    export function cloneNode<T extends Node>(node: T, location?: TextRange, flags?: NodeFlags, parent?: Node): T {
+    export function cloneNode<T extends Node>(node: T, location?: TextRange, flags?: NodeFlags, parent?: Node, original?: Node): T {

```

Comment:
It will. I'm staging the changes a PR at a time so as no to introduce a significant amount of churn as part of the review.